packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -2052,6 +2052,19 @@ abstract class BuildContext {
   /// managing the rendering pipeline for this context.
   BuildOwner get owner;
 
+  /// Determines if [widget] is in the process of updating.
+  ///
+  /// This flag will return `true` if [widget] is in one of those life-cycles:
+  /// - [StatelessWidget.build]
+  /// - [State.build]
+  /// - [RenderObjectWidget.createRenderObject]
+  /// - [RenderObjectWidget.updateRenderObject]
```

Comment:
It would help to add documentation here explaining why these life-cycles are similar enough to be grouped this way, particularly for implementers of new elements (or for future us when we realize we missed some case and want to remember what made these special).

Maybe something like:

> This flag is true whenever the tree is rebuilding this specific build context. When this flag is true, the element is guaranteed to be [mounted], and is in the process of going from a dirty or initial state to an updated state based on the configuration of its [widget].

Reply:
Agreed.
The common point between all of these would be:
- they update a tree declaratively
- they can use InheritedWidgets

Would that be enough of a description? I'm not too sure what else to put here

Reply:
Sure.  As long as it's following the style guide and sufficiently detailed it should be fine.

Reply:
About the overall comments on the added description:

I really do not know what to say here.
In theory, it should just be "build". The RenderObject life-cycles are here just because they are the equivalent to "build" for that widget.

I do not know how to describe it otherwise, that explain we purposefully do not want to include `initState`/`didChangeDependencies`.